File: testdata/bench/go/BenchmarkTest00072.go
package bench

import (
	"io"
	"net/http"
	"net/url"
)

// URL parse + host allowlist
func Handler00072(w http.ResponseWriter, r *http.Request) {
	target := r.URL.Query().Get("url")
	parsed, err := url.Parse(target)
	if err != nil || parsed.Host != "api.trusted.com" {
		http.Error(w, "forbidden", 403)
		return
	}
	resp, _ := http.Get(parsed.String())
	defer resp.Body.Close()
	io.Copy(w, resp.Body)
}

File: testdata/bench/rust/BenchmarkTest00072.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: Integer ID prevents SSRF
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("id").unwrap_or(&String::new()).clone();
    let id: u64 = match param.parse() {
        Ok(v) => v,
        Err(_) => return HttpResponse::BadRequest().body("invalid id"),
    };
    let url = format!("https://api.example.com/items/{}", id);
    let resp = reqwest::get(&url).await.unwrap();
    HttpResponse::Ok().body(resp.text().await.unwrap_or_default())
}
